mii/api.py

```diff
@@ -0,0 +1,149 @@
+                                **kwargs) -> MIIConfig:
+    # Parse all model_config kwargs
+    model_config, remaining_kwargs = _parse_kwargs_to_model_config(model_name_or_path=model_name_or_path, model_config=model_config, **kwargs)
+
+    if mii_config is None:
+        mii_config = {}
+
+    assert isinstance(mii_config, dict), "mii_config must be a dict"
+
+    # Verify that any model_config kwargs match any existing model_config in the mii_config
+    if "model_config" in mii_config:
+        assert mii_config.get("model_config") == model_config, "mii_config['model_config'] must match model_config"
+    else:
+        mii_config["model_config"] = model_config
+
+    # Fill mii_config dict with relevant kwargs, raise error on unknown kwargs
+    for key, val in remaining_kwargs.items():
+        if key in MIIConfig.__dict__["__fields__"]:
+            if key in mii_config:
+                assert mii_config.get(key) == val, f"{key} in mii_config must match {key}"
+            mii_config[key] = val
+        else:
+            raise UnknownArgument(f"Keyword argument {key} not recognized")
+
+    # Return the MIIConfig object
+    mii_config = MIIConfig(**mii_config)
+    return mii_config
+
+
+def client(model_or_deployment_name: str) -> MIIClient:
+    mii_config = get_mii_config(model_or_deployment_name)
+
+    return MIIClient(mii_config)
+
+
+def serve(model_name_or_path: str = "",
+          model_config: Optional[Dict[str,
+                                      Any]] = None,
+          mii_config: Optional[Dict[str,
+                                    Any]] = None,
+          **kwargs) -> Union[None,
+                             MIIClient]:
+    mii_config = _parse_kwargs_to_mii_config(model_name_or_path=model_name_or_path,
+                                             model_config=model_config,
+                                             mii_config=mii_config,
+                                             **kwargs)
+
+    #MIIServer(mii_config)
+    create_score_file(mii_config)
+
+    if mii_config.deployment_type == DeploymentType.LOCAL:
+        import_score_file(mii_config.deployment_name, DeploymentType.LOCAL).init()
+        return MIIClient(mii_config=mii_config)
+    if mii_config.deployment_type == DeploymentType.AML:
+        acr_name = mii.aml_related.utils.get_acr_name()
+        mii.aml_related.utils.generate_aml_scripts(
+            acr_name=acr_name,
+            deployment_name=mii_config.deployment_name,
+            model_name=mii_config.model_config.model,
+            task_name=mii_config.model_config.task,
+            replica_num=mii_config.model_config.replica_num,
+            instance_type=mii_config.instance_type,
+            version=mii_config.version,
+        )
+        print(
+            f"AML deployment assets at {mii.aml_related.utils.aml_output_path(mii_config.deployment_name)}"
+        )
+        print("Please run 'deploy.sh' to bring your deployment online")
```

Comment:
It would be good to have a recommended method to output the log, logger or print.
We may not need to work on it in this PR, let me just share the idea before I forget.